x/crosschain/keeper/evm_hooks.go

```diff
@@ -305,14 +305,25 @@ func (k Keeper) ProcessZetaSentEvent(
 
 // ValidateZRC20WithdrawEvent checks if the ZRC20Withdrawal event is valid
 // It verifies event information for BTC chains and returns an error if the event is invalid
-func (k Keeper) ValidateZRC20WithdrawEvent(ctx sdk.Context, event *zrc20.ZRC20Withdrawal, chainID int64) error {
+func (k Keeper) ValidateZRC20WithdrawEvent(
+	ctx sdk.Context,
+	event *zrc20.ZRC20Withdrawal,
+	chainID int64,
+	coinType coin.CoinType,
```

Comment:
Maybe we can pss the foreign coin object instead of these two?

Reply:
https://github.com/zeta-chain/node/pull/3374/commits/a409a1a011a7e13685bada4e062f69d63467f7d7